### PR TITLE
docs(lpeg): merge upstream changes

### PR DESCRIPTION
**Pull request by MariaSolOs**
Adds the changes merged upstream: https://github.com/LuaCATS/lpeg/pull/9

Follow-up from some typos and code style nits I found in https://github.com/neovim/neovim/pull/26989.

### CODE REVIEW

**Comment by wookayin** [comment]
docs rather than chore?

File: runtime/doc/lua.txt [hunk 1]
```diff
@@ -3321,19 +3321,18 @@ vim.lpeg.setmaxstack({max})                           *vim.lpeg.setmaxstack()*
       • {max}  (`integer`)
 
 vim.lpeg.type({value})                                       *vim.lpeg.type()*
-    Returns the string `"pattern"` if the given value is a pattern, otherwise
-    `nil`.
+    Returns the string 'pattern `if the given value is a pattern, otherwise` nil`.
 
     Return: ~
-        (`"pattern"?`)
+        ('pattern ?`)
```
Comment (wookayin) [inline runtime/doc/lua.txt:3327]: I think this is meant to be the literal string `"pattern"`, not a type `pattern`.

**Comment by MariaSolOs** [comment]
> docs rather than chore?

I'll use `build` since I was told that's our `chore` ;)

**Comment by clason** [comment]
I think in this case "docs" is fine ;)

File: runtime/doc/lua.txt (hunk 1 above)
Comment (MariaSolOs) [reply runtime/doc/lua.txt:3327]: Hmmm weird. This is indeed a literal in the metafile:
https://github.com/neovim/neovim/blob/d3d10ab25cc440a3f14dcb184d91ef5dc2526c8e/runtime/lua/vim/_meta/lpeg.lua#L91

Comment (wookayin) [reply runtime/doc/lua.txt:3327]: The docgen probably doesn't understand `'literal'` strings but only does `"literal"` strings... 😂

Comment (MariaSolOs) [reply runtime/doc/lua.txt:3327]: AHHHHHHH

Comment (wookayin) [reply runtime/doc/lua.txt:3327]: See this: https://github.com/neovim/neovim/pull/26989#discussion_r1450986576

File: runtime/lua/vim/_meta/lpeg.lua [hunk 2]
```diff
@@ -114,40 +114,40 @@ function vim.lpeg.setmaxstack(max) end
...
 --- @param pattern vim.lpeg.Pattern
 --- @return vim.lpeg.Pattern
 function vim.lpeg.B(pattern) end
 
 --- Returns a pattern that matches any single character belonging to one of the given ranges.
 --- Each `range` is a string `xy` of length 2, representing all characters with code between the codes of
---- `x` and `y` (both inclusive). As an example, the pattern `lpeg.R("09")` matches any digit, and
---- `lpeg.R("az", "AZ")` matches any ASCII letter.
+--- `x` and `y` (both inclusive). As an example, the pattern `lpeg.R('09')` matches any digit, and
+--- `lpeg.R('az', 'AZ')` matches any ASCII letter.
```
Comment (zeertzjq) [inline runtime/lua/vim/_meta/lpeg.lua:134]:
```suggestion
--- `x` and `y` (both inclusive). As an example, the pattern ``lpeg.R('09')`` matches any digit, and
--- ``lpeg.R('az', 'AZ')`` matches any ASCII letter.
```

File: runtime/lua/vim/_meta/lpeg.lua [hunk 3]
```diff
@@ -114,40 +114,40 @@ function vim.lpeg.setmaxstack(max) end
...
+--- assert(pattern:match('hello') == 6)
 --- ```
 ---
 --- @param ... string
 --- @return vim.lpeg.Pattern
 function vim.lpeg.R(...) end
 
 --- Returns a pattern that matches any single character that appears in the given string (the `S` stands for Set).
---- As an example, the pattern `lpeg.S("+-*/")` matches any arithmetic operator. Note that, if `s` is a character
+--- As an example, the pattern `lpeg.S('+-*/')` matches any arithmetic operator. Note that, if `s` is a character
```
Comment (zeertzjq) [inline runtime/lua/vim/_meta/lpeg.lua:148]:
```suggestion
--- As an example, the pattern ``lpeg.S('+-*/')`` matches any arithmetic operator. Note that, if `s` is a character
```

File: runtime/lua/vim/_meta/lpeg.lua [hunk 4]
```diff
@@ -114,40 +114,40 @@ function vim.lpeg.setmaxstack(max) end
...
 --- (that is, a string of length 1), then `lpeg.P(s)` is equivalent to `lpeg.S(s)` which is equivalent to
---- `lpeg.R(s..s)`. Note also that both `lpeg.S("")` and `lpeg.R()` are patterns that always fail.
+--- `lpeg.R(s..s)`. Note also that both `lpeg.S('')` and `lpeg.R()` are patterns that always fail.
```
Comment (zeertzjq) [inline runtime/lua/vim/_meta/lpeg.lua:150]:
```suggestion
--- `lpeg.R(s..s)`. Note also that both ``lpeg.S('')`` and `lpeg.R()` are patterns that always fail.
```

File: runtime/lua/vim/_meta/lpeg.lua (hunk 2 above)
Comment (MariaSolOs) [reply runtime/lua/vim/_meta/lpeg.lua:134]: Oh nvm, it's another doxygen thing.

**Comment by MariaSolOs** [comment]
AHHH I don't understand what's going on. I'm definitely running `make doc` and committing the changes but the CI check keeps failing :(

**Comment by wookayin** [comment]
Congratulations, you've just found a bug that happens only [on CI](https://github.com/neovim/neovim/actions/runs/7851523675/job/21428718967?pr=27402#step:5:4776).

Actually, this difference turns out to be due to different doxygen versions. I and you are probably using the latest version (1.10), but CI runs doxygen 1.9.1 (latest on ubuntu packages); with doxygen 1.9.1, I can reproduce the same diff locally. We'll need to bump up doxygen for CI!

**Comment by MariaSolOs** [comment]
> Congratulations, you've just found a bug that happens only [on CI](https://github.com/neovim/neovim/actions/runs/7851523675/job/21428718967?pr=27402#step:5:4776).
> 
> Actually, this difference turns out to be due to different doxygen versions. I and you are probably using the latest version (1.10), but CI runs doxygen 1.9.1 (latest on ubuntu packages); with doxygen 1.9.1, I can reproduce the same (incorrect) diff locally. We'll need to bump up doxygen for CI!

Yayyy do I get a prize :D ??

**Comment by wookayin** [comment]
I did a bit of bisect and can confirm that v1.9.5 (release date 26-08-2022) is the first version where this bug is fixed. https://github.com/doxygen/doxygen/releases

22.04 LTS (jammy): https://packages.ubuntu.com/source/jammy/doxygen installs v1.9.1 (release date 08-01-2021).

Anyway we can bump doxygen to latest: #27410

**Comment by MariaSolOs** [comment]
Update: I'm waiting on #27480 to get merged first since it fixes the doxygen issues (by removing it lol).

**Comment by clason** [comment]
> Update: I'm waiting on #27480 to get merged first since it fixes the doxygen issues (by removing it lol).

It got merged 🤗

**Review by clason — APPROVED** [review]

File: runtime/lua/vim/_meta/lpeg.lua [hunk 5]
```diff
@@ -113,40 +113,40 @@ function vim.lpeg.setmaxstack(max) end
...
 --- @param ... string
 --- @return vim.lpeg.Pattern
 function vim.lpeg.R(...) end
 
 --- Returns a pattern that matches any single character that appears in the given string (the `S` stands for Set).
---- As an example, the pattern `lpeg.S("+-*/")` matches any arithmetic operator. Note that, if `s` is a character
+--- As an example, the pattern ``lpeg.S('+-*/')`` matches any arithmetic operator. Note that, if `s` is a character
 --- (that is, a string of length 1), then `lpeg.P(s)` is equivalent to `lpeg.S(s)` which is equivalent to
---- `lpeg.R(s..s)`. Note also that both `lpeg.S("")` and `lpeg.R()` are patterns that always fail.
+--- `lpeg.R(s..s)`. Note also that both ``lpeg.S('')`` and `lpeg.R()` are patterns that always fail.
```
[the remark this answers is not in the document]
Comment (MariaSolOs) [reply runtime/lua/vim/_meta/lpeg.lua:149]: Fixed with #27659

[the remark this answers is not in the document]
Comment (shahardra) [reply runtime/lua/vim/_meta/lpeg.lua:149]: How

**Pull request merged**